aiohttp/web_request.py

```diff
@@ -151,15 +181,60 @@ def secure(self):
         return self.url.scheme == 'https'
 
     @reify
+    def forwarded(self):
+        """ A frozendict containing parsed Forwarded header(s).
+
+        Makes an effort to parse Forwarded headers as specified by RFC 7239:
+
+        - It adds all parameters (by, for, host, proto) in the order it finds
+          them; starting at the topmost / first added 'Forwarded' header, at
+          the leftmost / first-added parwameter.
+        - It checks that the value has valid syntax in general as specified in
+          section 4: either a 'token' or a 'quoted-string'.
+        - It un-escapes found escape sequences.
+        - It does NOT validate 'by' and 'for' contents as specified in section
+          6.
+        - It does NOT validate 'host' contents (Host ABNF).
+        - It does NOT validate 'proto' contents for valid URI scheme names.
+
+        Returns a dict(by=tuple(...), for=tuple(...), host=tuple(...),
+        proto=tuple(...), ) 
+        """
+        params = {'by': [], 'for': [], 'host': [], 'proto': []}
```

Comment:
I believe the property's type should be `MultyDict` instead of dict of lists.
It preserves all information about proxies in right order but original host still could be checked as `req.forwarded.get('host')`